gobblin-core/src/main/java/gobblin/writer/PartitionedDataWriter.java

```diff
@@ -171,4 +172,24 @@ public void close() throws IOException {
     return this.builder.get().forPartition(partition).withWriterId(this.baseWriterId + "_" + this.writerIdSuffix++)
         .build();
   }
+
+  @Override
+  public State getFinalState() {
+
+    State state = new State();
+    try {
+      for (Map.Entry<GenericRecord, DataWriter<D>> entry : this.partitionWriters.asMap().entrySet()) {
+        if (entry.getValue() instanceof FinalState) {
+          state.addAll(((FinalState)entry.getValue()).getFinalState());
```

Comment:
Unfortunately there is no consistency in order of partitions, so the suffix _0 does not mean anything. Can we just serialize the values of the partition (the partition is an avro record, so it shouldn't be hard)?